stingray/fourier.py

```diff
@@ -0,0 +1,1286 @@
+    common_mean = Nph / N
+
+    if common_variance is not None:
+        # Note: the variances we summed were means, not sums. Hence M, not M*N
+        common_variance /= M
+
+    # Transform a sum into the average
+    unnorm_cross = cross / M
+
+    # Final normalization (If not done already!)
+    if use_common_mean:
+        cross = normalize_periodograms(
+            unnorm_cross, dt, N, common_mean, norm=norm, variance=common_variance
+        )
+
+    results = Table()
+    results["freq"] = freq
+    results["power"] = cross
+    results["unnorm_power"] = unnorm_cross
+    results.meta.update({"n": N, "m": M, "dt": dt,
+                         "norm": norm,
+                         "df": 1 / (dt * N),
+                         "nphots": Nph,
+                         "mean": common_mean,
+                         "segment_size": dt * N})
+
+    return results
+
+
+def avg_cs_from_iterables_quick(
+    flux_iterable1,
+    flux_iterable2,
+    dt,
+    norm="abs"
+):
+    """Like `avg_cs_from_iterables`, with default options that make it quick.
+
+    Assumes that:
+
+    * the flux iterables return counts, no other units
+    * the mean is calculated over the whole light curve, and normalization
+      is done at the end
+    * no auxiliary PDSs are returned
+    * only positive frequencies are returned
+    * the spectrum is complex, no real parts or absolutes
+    * no progress bars
+
+    Parameters
+    ----------
+    flux_iterable1 : `iterable` of `np.array`s or of tuples (`np.array`, `np.array`)
+        Iterable providing either equal segments of light curve, or of light curve
+        and errors. They must all be of the same length.
+    flux_iterable2 : `iterable` of `np.array`s or of tuples (`np.array`, `np.array`)
+        Same as ``flux_iterable1``, for the reference channel
+    dt : float
+        Time resolution of the light curves used to produce periodograms
+
+    Other Parameters
+    ----------------
+    norm : str, default "abs"
+        The normalization of the periodogram. "abs" is absolute rms, "frac" is
+        fractional rms, "leahy" is Leahy+83 normalization, and "none" is the
+        unnormalized periodogram
+
+    Returns
+    -------
+    freq : `np.array`
+        The periodogram frequencies
+    power : `np.array`
+        The normalized cross spectral powers
+    N : int
+        the number of bins in the light curves used in each segment
+    M : int
+        the number of averaged periodograms
+    mean : float
+        the mean flux (geometrical average of the mean fluxes in the two channels)
+
+    """
+    # Initialize stuff
+    unnorm_cross = unnorm_pds1 = unnorm_pds2 = None
+    M = 0
+
+    sum_of_photons1 = sum_of_photons2 = 0
+
+    for flux1, flux2 in zip(flux_iterable1, flux_iterable2):
+        if flux1 is None or flux2 is None:
+            continue
+
+        N = flux1.size
+
+        # Calculate the sum of each light curve, to calculate the mean
+        # This will
```

Comment:
This will ...? You can't leave me hanging like this! 😆

Reply:
Ooops